### PR TITLE
Make MatchKey method __getitem__ more useful

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

**Comment by jafingerhut** [comment]
Also factor out a method to convert abbreviated field names to full field names.

File: p4runtime_sh/shell.py [hunk 1]
```diff
@@ -340,23 +340,27 @@ def _ipython_key_completions_(self):
...
     def __setitem__(self, name, value):
         field_info = self._get_mf(name)
         self._mk[name] = self._parse_mf(value, field_info)
         print(self._mk[name])
 
     def __getitem__(self, name):
-        _ = self._get_mf(name)
-        print(self._mk.get(name, "Unset"))
+        fullname = self._full_field_name(name)
+        return self._mk.get(fullname, "Unset")
```
Comment (antoninbas) [inline p4runtime_sh/shell.py:363]: while I agree that returning a value makes more sense, I feel like `"Unset"` makes more sense as something you display to the user in the context of the shell, and not as a return value (I feel like `None` would be better there), so I suggest something like this:
```python
    def __getitem__(self, name):
        fullname = self._full_field_name(name)
        f = self._mk.get(fullname, None)
        if f is None:
            print("Unset")
        return f
```

**Review by antoninbas — COMMENTED** [review]
I believe a similar situation exists for action parameters: https://github.com/p4lang/p4runtime-shell/blob/4bc4cd5c6b86a9669e5d6102da5c86ff0c0917ce/p4runtime_sh/shell.py#L595-L597

File: p4runtime_sh/shell.py (hunk 1 above)
Comment (jafingerhut) [reply p4runtime_sh/shell.py]: Agreed, and done.

File: p4runtime_sh/shell.py [hunk 2]
```diff
@@ -340,23 +340,27 @@ def _ipython_key_completions_(self):
...
             "'{}' is not a valid match field name, nor a valid unique suffix, "
             "for table '{}'".format(name, self._table_name))
 
+    def _get_mf(self, name):
+        fullname = self._full_field_name(name)
+        return self._fields[fullname]
+
     def __setitem__(self, name, value):
         field_info = self._get_mf(name)
         self._mk[name] = self._parse_mf(value, field_info)
```
[the remark this answers is not in the document]
Comment (jafingerhut) [reply p4runtime_sh/shell.py]: Good catch.  Done.

**Review by antoninbas — APPROVED** [review]
@jafingerhut I'm merging this now, it would be good if you could do the same change for action parameters

**Comment by jafingerhut** [comment]
@antoninbas "I'm merging this now, it would be good if you could do the same change for action parameters"

Can you explain a little bit more what you mean here?  Action parameter names cannot have dots in the middle, so I don't understand how I could make a similar change for action parameters.

**Comment by jafingerhut** [comment]
@antoninbas Please disregard my previous comment.  I was confused.  You are asking me to make `__getitem__` work so that expressions like `te.action['field_name']` return a useful value.  Makes sense.

**Pull request merged**